sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala

```diff
@@ -71,6 +71,12 @@ object HiveThriftServer2 extends Logging {
   }
 
   def main(args: Array[String]) {
+    // If the arguments contains "-h" or "--help", print out the usage and exit.
+    if (args.contains("-h") || args.contains("--help")) {
+      HiveServer2.main(args)
```

Comment:
This is a bit ugly, but it is the only solution I can think of.

Reply:
That's reasonable. But just return in the line below. System.exit is unnecessarily abrupt.